ddtrace/contrib/gevent/greenlet.py

```diff
@@ -1,33 +1,30 @@
 import gevent
-import gevent.pool as gpool
 
-from .provider import GeventContextProvider
+from ddtrace._trace.provider import _DD_CONTEXTVAR
 
 
 GEVENT_VERSION = gevent.version_info[0:3]
 
 
 class TracingMixin(object):
     def __init__(self, *args, **kwargs):
-        # get the active context/span if available
-        current_g = gevent.getcurrent()
-        ctx = getattr(current_g, GeventContextProvider._CONTEXT_ATTR, None)
-
-        # create the Greenlet as usual
+        # Storse the current Datadog context.
+        # This is necessary to ensure tracing context is passed to greenlets.
+        # Avoids setting Greenlet.gr_context, setting field could introduce
+        # unintended side-effects in third party libraries.
+        self.trace_context = _DD_CONTEXTVAR.get()
         super(TracingMixin, self).__init__(*args, **kwargs)
 
-        # copy the active span/context into the new greenlet
-        if ctx:
-            setattr(self, GeventContextProvider._CONTEXT_ATTR, ctx)
+    def run(self):
+        # Propagates Datadog context to spawned greenlets
+        _DD_CONTEXTVAR.set(self.trace_context)
+        super(TracingMixin, self).run()
```

Comment:
why do we not restore things to the parent context after run ?

Reply:
For context, this is not an issue right now, I was mostly curious if this made sense.